bayes/src/main/java/com/enigmastation/ml/bayes/impl/FisherClassifierImpl.java

```diff
@@ -79,26 +91,23 @@ protected double featureProb(Object feature, Object category) {
                 .stream()
                 .mapToDouble(c -> super.featureProb(feature, c))
                 .sum();
-//        for (Object c : categories()) {
-//            frequencySum += super.featureProb(feature, c);
-//        }
         return clf / frequencySum;
     }
 
     protected double fisherProbability(Object source, Object category) {
-        final double p[] = {1.0};
+        final double[] p = {1.0};
         List<Object> features = getFeatures(source);
         features
                 .stream()
                 .mapToDouble(f -> weightedProb(f, category))
                 .forEach(w -> p[0] *= w);
         double fisherScore = -2.0 * Math.log(p[0]);
-        return invChi(fisherScore, features.size() * 2);
+        return invChi(fisherScore, ((double) features.size()) * 2);
```

Comment:
Why are we concerned over a very ordinary conversion? Why not change to 2.0 instead of 2, rather than double the complexity of the line?